lib/workers/repository/cache.ts

```diff
@@ -57,7 +57,7 @@ async function generateBranchCache(
     let isModified = false;
     let isBehindBase = false;
     if (sha) {
-      parentSha = await getBranchParentSha(branchName);
+      parentSha = getCachedBranchParentShaResult(branchName, sha);
```

Comment:
How do you know it's *cached* at this point?

Reply:
During branch-creation it has to be there as we just added it.
For all other cases we will get either a string or null
Null if `isModified=true` and we deleted it. It will continue to remian `null` till a new commit is added.
String in all other cases as syncing branchState has no effect on `parentSha`.

One thing to note is that it can be `null` too in the cache.